src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleasePublishingStatusRepository.cs

```diff
@@ -47,5 +48,38 @@ public Task<IEnumerable<ReleasePublishingStatus>> GetAllByOverallStage(Guid rele
             var query = new TableQuery<ReleasePublishingStatus>().Where(filter);
             return _publisherTableStorageService.ExecuteQueryAsync(PublisherReleaseStatusTableName, query);
         }
+
+        public async Task RemovePublisherReleaseStatuses(List<Guid> releaseIds)
+        {
+            if (releaseIds.IsNullOrEmpty())
+            {
+                // If filter is string.Empty, the query returns all table entities
+                return;
```

Comment:
If an empty filter is considered an error, should we feed something back to the caller so that it can be handled accordingly, such as an `ArgumentNullException(nameof(releaseIds))`, or a failed/cancelled task?

Reply:
I think it is correct as it is. It's possible for `DeleteReleasesForPublications` to be called on publications that have no releases. In those cases, we'd expect `RemovePublisherReleaseStatuses` to be called with an empty array and to handle that correctly i.e. do nothing.

I've updated the comment to (hopefully) make it clearer the reason why this initial if statement is necessary.